aspnet/dnx/commands.rst

```diff
@@ -1,8 +1,121 @@
+
+Using commands in your project
+------------------------------
+
+In the ``commands`` section of the below *project.json* example, four commands are listed:
+
+.. code-block:: json
+
+    {
+      "version": "1.0.0",
+      "webroot": "wwwroot",
+      "exclude": [
+        "wwwroot"
+      ],
+      "dependencies": {
+        "Kestrel": "1.0.0-beta4",
+        "Microsoft.AspNet.Diagnostics": "1.0.0-beta4",
+        "Microsoft.AspNet.Hosting": "1.0.0-beta4",
+        "Microsoft.AspNet.Server.IIS": "1.0.0-beta4",
+        "Microsoft.AspNet.Server.WebListener": "1.0.0-beta4",
+        "Microsoft.AspNet.StaticFiles": "1.0.0-beta4"
+      },
+      "commands": {
+		"web": "Microsoft.AspNet.Hosting --server Microsoft.AspNet.Server.WebListener --server.urls http://localhost:5000",
+		"kestrel": "Microsoft.AspNet.Hosting --server Kestrel --server.urls http://localhost:5001",
+		"gen": "Microsoft.Framework.CodeGeneration",
+		"ef": "EntityFramework.Commands"
+      },
+      "frameworks": {
+        "dnx451": { },
+        "dnxcore50": { }
+      }
+    }
+
+The commands can include a set of arguments that will be passed to the DNX. In the above example, the first part of a command statement is an assembly with an entry point that the DNX will try to execute. Notice that in the ``commands`` section shown above, the ``ef`` assembly doesn't require any extra arguments, so all that is needed to define the command is the name of the assembly. For the ``kestrel`` command, there are three arguments. The first is the needed assembly, the second is the ``server`` agrument, and the third is the ``server.urls`` argument. The ``kestrel`` command will check the ``Microsoft.AspNet.Hosting`` assembly for an entry point, then it will pass the ``server`` and ``serveral.urls`` parameters to the entry point. 
+
+.. note:: The assembly listed in the ``commands`` section should be pulled in by a package that your application depends on.
+
+Installing and creating commands
+--------------------------------
+
+You can add additional commands to your project by modifying the *project.json* file. Additionally, you can use the DNU to install commands globally on a machine using the following syntax::
+
+    dnu commands install MyCommand
+	
+For more details about installing commands, see `DNX Utility <https://github.com/aspnet/Home/wiki/DNX-utility>`_.	
+
+Executing commands
+------------------
+
+To execute commands, you could run them from the command prompt, but a faster way when working with VS Code is to open the **Command Palette** (Ctrl+Shift+P), start typing the name of the command you want to run, and press **Enter**.
+
+You can use DNX to execute the commands defined by your project by entering the following in the command prompt from your project's directory::
+
+    dnx . kestrel
+
+In VS Code, the same command would appear as follows::	
```

Comment:
I think you can remove this text through to the end of this section. Instead it would be helpful to explain how running a command is just a short hand for specifying the command assembly and it's arguments directly to DNX. We should also show how to use DNX to run commands from a project in a different directory.